src/main/java/org/isf/opd/gui/OpdEditExtended.java

```diff
@@ -1601,29 +1606,36 @@ private JButton getOkButton() {
 			okButton = new JButton(MessageBundle.getMessage("angal.common.ok.btn"));
 			okButton.setMnemonic(MessageBundle.getMnemonic("angal.common.ok.btn.key"));
 			okButton.addActionListener(actionEvent -> {
+				
+				if (opdDateFieldCal.getDate() != null) {
+					visitDateOpd = new GregorianCalendar();
+					visitDateOpd.setTime(opdDateFieldCal.getDate());
+					opd.setVisitDate(visitDateOpd);
+					opd.setDate(visitDateOpd);
+				} else {
+					opd.setVisitDate(visitDateOpd);
+					opd.setDate(null);
+				}
 				int opdProgYear = 0;
-				if (!jOpdNumField.getText().equals("") || !jOpdNumField.getText().contains(" ")) {
-					GregorianCalendar gregDate = new GregorianCalendar();
-					gregDate.setTime(opdDateFieldCal.getDate());
+				
+				if (jOpdNumField.isEditable()) {
 					try {
 						opdProgYear = Integer.parseInt(jOpdNumField.getText());
-					} catch (NumberFormatException e) {
-						MessageDialog.error(OpdEditExtended.this, "angal.opd.opdnumbermustbeanumber.msg");
-						return;
-					}
-				}
-				if (insert || opd.getProgYear() != opdProgYear) {
-					try {
-						GregorianCalendar gregDate = new GregorianCalendar();
-						if (opdManager.isExistOpdNum(opdProgYear, gregDate.get(Calendar.YEAR))) {
+						if (opdManager.isExistOpdNum(opdProgYear, visitDateOpd.get(Calendar.YEAR))) {
 							MessageDialog.error(OpdEditExtended.this, "angal.opd.opdnumberalreadyexist.msg");
-							jOpdNumField.setText(getOpdProgYear());
+							if (insert) jOpdNumField.setText(String.valueOf(getOpdProgYear(visitDateOpd)));
```

Comment:
For consistency there should be braces here.

Reply:
Fixed, thanks!